openweather-based-bidding/main.gs

```diff
@@ -35,86 +38,92 @@ function monitorWeatherAndSyncWithDV360() {
-    const currentWindspeed          = currentWeather.wind_speed;
+    row[config.get('col-wind-curr')] = currentWeather.wind_speed;
+
+    // Initialize all conditions as satisfied
+    const satisfiedConditions = {
+      tempMin: true,
+      tempMax: true,
+      precip: true,
+      windSpeed: true,
+    };
 
     // Check if min temperature condition is satisfied
-    let tempConditionSatisfied = true;
-    if (tempMin || tempMin !== '') {
-      tempConditionSatisfied = (tempMin <= currentTemperature);
+    if (tempMin) {
+      satisfiedConditions.tempMin = tempMin <= row[config.get('col-temp-curr')];
     }
 
     // Check if max temperature condition is satisfied
-    if (tempMax || tempMax !== '') {
-      tempConditionSatisfied = tempConditionSatisfied
-        && (currentTemperature <= tempMax);
+    if (tempMax) {
+      satisfiedConditions.tempMax = row[config.get('col-temp-curr')] <= tempMax;
     }
 
     // Check if rain/snow condition is satisfied
-    let rainSnowConditionSatisfied = true;
     if (onlyIfRainingOrSnow) {
-      rainSnowConditionSatisfied = currentRainSnowCondition;
+      satisfiedConditions.precip = row[config.get('col-precip-curr')];
     }
 
     // Check if wind speed condition is satisfied
-    let windConditionSatisfied = true;
     if (windMin) {
-      windConditionSatisfied = (currentWindspeed >= windMin);
+      satisfiedConditions.windSpeed = row[config.get('col-wind-curr')] >= windMin;
     }
 
     // Check if all conditions are satisfied
-    const activate = rainSnowConditionSatisfied &&
-                      tempConditionSatisfied &&
-                      windConditionSatisfied;
-
-    // Log status to debug panel
-    Logger.log(
-      `Updating: IO: ${insertionOrderId}, LI: ${lineItemId}, `
-      + `Active: ${activate}, `
-      + `Temp Min.: ${tempMin}, Temp Max.: ${tempMax}, `
-      + `Curr. Temp.: ${currentTemperature}, `
-      + `Curr. Rain/Snow: ${currentRainSnowCondition}, `
-      + `Wind Min: ${windMin}, Curr. Wind speed: ${currentWindspeed}`
-    );
-
-    /*
-     * This does the API request to change the LI status.
-     * If you wish to change the IO status instead just change to:
-     * dv360.switchIOStatus(advertiserId, insertionOrderId, activate)
-     */
+    const activate = allObjectPropertiesTrue(satisfiedConditions);
+
+    // Switch Line Item status
     dv360.switchLIStatus(advertiserId, lineItemId, activate);
 
-    // Log the status to the spreadsheet
-    const row = i + 1;
-    sheet.getRange( 'J' + row ).setValue( currentTemperature );
-    sheet.getRange( 'K' + row ).setValue( currentRainSnowCondition );
-    sheet.getRange( 'L' + row ).setValue( currentWindspeed );
-    sheet.getRange( 'M' + row ).setValue( activate ? 'Active' : 'Paused' );
-    sheet.getRange( 'N' + row ).setValue( new Date().toISOString() );
+    // Switch Insertion Order status
+    dv360.switchIOStatus(advertiserId, insertionOrderId, activate)
```

Comment:
We should switch only one entity at a time (either IO or LI). I suggest the following logic:
1. if IO id is defined we switch `dv360.switchIOStatus(advertiserId, insertionOrderId, activate)`
2. else (if IO is not defined), but LI id is defined, then `dv360.switchLIStatus(advertiserId, lineItemId, activate);`